### ISSUE TITLE
Poetry 1.4.0 can't install dependencies: Connection pool is full

### ISSUE THREAD

**Issue by borice** [python-poetry/poetry#7584]
With the upgrade to Poetry 1.4.0 I am now getting the following error when running `poetry install` in CircleCI Enterprise.
> Connection pool is full, discarding connection: pypi.org. Connection pool size: 10 

I have changed the build script to install Poetry 1.3.2, and the error went away.
I have checked that going back to 1.4.0 brings back the error also.

It seems that whatever changed between 1.3.2 and 1.4.0 is now causing this error for us in CircleCI.

**Comment by dimbleby** [comment]
that's just a benign warning.  You haven't said what your problem is, but I don't think it's this.

**Comment by borice** [comment]
Well, this is what I was seeing in CircleCI:
![image](https://user-images.githubusercontent.com/431091/222254775-cd481cbf-cd04-4017-8528-47ea77fc8d34.png)

... but after reverting to 1.3.2 and restarting the build, I ran into this other error in the Docker image building step:

![image](https://user-images.githubusercontent.com/431091/222255021-ea228998-05be-407f-a61b-f3c77ca290c3.png)

I reverted that setup to use 1.3.2 also, and error was gone.  Went back to 1.4.0 briefly to test if this was a sporadic error, and error came back.  Back again to 1.3.2 error went away again.  So 1.4.0 seems to have some regression issue.

**Comment by dimbleby** [comment]
duplicate #7572, please close

**Comment by allanlewis** [comment]
I don't think this is related to #7572, it's the issue reported in [this SO post](https://stackoverflow.com/questions/74385209/poetry-install-throws-connection-pool-is-full-discarding-connection-pypi-org). The solution suggested there, which works for me, is to set the Poetry config setting `installer.max-workers` to 10.
